src/rgw/rgw_swift_auth.cc

```diff
@@ -304,7 +304,7 @@ ExternalTokenEngine::authenticate(const std::string& token,
   }
 
   std::string auth_url = g_conf->rgw_swift_auth_url;
-  if (auth_url[auth_url.length() - 1] != '/') {
+  if (auth_url.back() != '/') {
```

Comment:
May we have a comment saying that`ExternalTokenEngine::is_applicable` guarantees `rgw_swift_auth_url` is non-empty?

Anyway, it's only a suggestion. Even now the commit is perfectly mergeable.